lib/pbench/server/api/resources/users_api.py

```diff
@@ -298,8 +299,22 @@ def __init__(self, logger, auth):
         self.logger = logger
         self.auth = auth
 
+    def operation_permitted(self, target_username: str) -> User:
+        """
+        Helper function to determine whether the API call is permitted for the target username
+        Right now it is only permited for an admin user and the target user itself
+        :param target_username: str
+        :return: target User object/ None
+        """
+        current_username = self.auth.token_auth.current_user().username
+        current_user = User.query(username=current_username)
+        target_user = User.query(username=target_username)
+        if current_user.is_admin() or current_username == target_username:
+            return target_user
```

Comment:
This check could be inverted to return `None` and moved up before doing the `query` on `target_username`, saving a small amount of time on the failure case.

Reply:
That is very good point @webbnh, Thanks

Reply:
I believe that there are actually several places in the code where we pull the current user from the auth token and then use it to query the database to get some attribute of the user -- they should probably all be "improved".

Speaking of which, I have a (possibly) better code suggestion:
```
        current_user = self.auth.token_auth.current_user()
        if current_user.username == target_username:
            return current_user
        if current_user.is_admin(): 
            return User.query(username=target_username)
        return None
```
It's slightly more complex (we're back to six lines...), but it avoids a DB query when we already know the answer (and at worst it's the same number of operations).